src/pkynetics/technique_analysis/dilatometry.py

```
@@ -0,0 +1,464 @@
+
+    min_points = 5
+    if np.sum(before_mask) < min_points or np.sum(after_mask) < min_points:
+        raise ValueError(
+            f"Insufficient points for fitting. Need at least {min_points} points in each region."
+        )
+
+    try:
+        before_fit = np.polyfit(temperature[before_mask], strain[before_mask], 1)
+        after_fit = np.polyfit(temperature[after_mask], strain[after_mask], 1)
+    except np.linalg.LinAlgError:
+        raise ValueError("Unable to perform linear fit on the data segments")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)